api-put-bucket.go

```diff
@@ -163,7 +163,8 @@ func (c Client) SetBucketPolicy(bucketName string, objectPrefix string, bucketPo
 	}
 
 	policyInfo, err := c.getBucketPolicy(bucketName)
```

Comment:
That is exactly what we don't want. All callers need not check for example Core API shouldn't check.

It is a change to reflect a core API should do lower implementation and return all errors. Special interpretation is left to the caller.